api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -76,7 +76,7 @@
    */
   String MAX_CONCURRENT_FILE_GROUP_REWRITES = "max-concurrent-file-group-rewrites";
 
-  int MAX_CONCURRENT_FILE_GROUP_REWRITES_DEFAULT = 1;
+  int MAX_CONCURRENT_FILE_GROUP_REWRITES_DEFAULT = 5;
```

Comment:
I support the idea of changing this default as we ask our users to change it in 9/10 cases.
@RussellSpitzer, any thoughts on this as you wrote this part?

Reply:
I like 5 but what I'd really like us to do is check the size of our "file groups" then use that to determine what to set max concurrent groups too. For example if we see all file groups are less than say target file size, we set Max Conncurrent = Num Cores

Reply:
We would probably need to look at the scheduler implementation (e.g. FIFO) and total resources available in the cluster vs the amount of data we need to compact and partitions. I think it makes sense exploring but if it becomes too tricky, I'd just default it to something more reasonable.

Reply:
> I support the idea of changing this default as we ask our users to change it in 9/10 cases.

True. Weekly once or twice, I am suggesting users configure it in Iceberg slack. Most of them doesn't read javadoc :( 
I would love to see this default value getting changed.

+1 for this change.

Reply:
I'm sold for this right now, then something more complicated later